src/battle_game/core/units/magic_tank.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+#include "battle_game/core/unit.h"
+#include "battle_game/core/units/tiny_tank.h"
+namespace battle_game::unit {
+class MagicTank : public Tank {
+ public:
+  MagicTank(GameCore *game_core, uint32_t id, uint32_t player_id);
+  void Render() override;
+  void Update() override;
+  void Protect();
+  void Death();
+  void ProtectClick();
+  void DeathClick();
+  void Magic();
```

Comment:
These functions should not be public, private or protected is enough

Reply:
Already change the position of those functions. Please approve it..